superset-frontend/src/components/ListView/LegacyFilters.tsx

```diff
@@ -67,9 +63,10 @@ export const FilterMenu = ({
           <MenuItem
             key={ft.id}
             eventKey={ft}
-            onSelect={(fltr: typeof ft) =>
-              setInternalFilters([...internalFilters, fltr])
-            }
+            // @ts-ignore
```

Comment:
What is the error?

Reply:
> Type 'MouseEvent<{}, MouseEvent>' is missing the following properties from type '{ Header: string; id: string; value: undefined; }': Header, id, value


Inconsistent callback function signature. onSelect in MenuItem expects an event, but the callback seems to be expecting the fltr type (bad variable name btw).

Didn't want to change the actual code and didn't dig deep too deep into how to make it proper, since this file is going to be deprecated anyway (@nytai is working on a new ListView).

Reply:
Yes, this is old code that's going to be killed soon. I think react-bootstrap was missing proper typings when this was being implemented so some type is probably being overridden incorrectly.

Reply:
The issue is the typings for react-bootstrap. If EventKey is present the callback will be called with that and not a MouseEvent, but the typings don't seem to account for that. This code is all going away soon, anyway, I wouldn't spend too much time on it